flink-python/pyflink/fn_execution/boot.py

```diff
@@ -0,0 +1,352 @@
+
+
+def pip_install_package(files, dir, name, force, optional, extras):
+    """
+    Pip install the specified python package.
+
+    :param files: The list of the downloaded file names.
+    :type files: list[str]
+    :param dir:  The path of the python package directory.
+    :type dir: str
+    :param name:  The file name of the python package.
+    :type name: str
+    :param force: Force install regardless of whether the package already installed.
+    :type force: bool
+    :param optional: If the specified python package not exists and this param is True,
+                     return 0. If the specified python package not exists and this param
+                     is False, raise an Exception.
+    :type optional: bool
+    :param extras: Specifies the extras_requires param of pip.
+    :type extras: list[str]
+    :return: 0 if install success or no need to install.
+    """
+    for filename in files:
+        if filename == name:
+            package = name
+            if extras is not None and len(extras) > 0:
+                package += "[" + ",".join(extras) + "]"
+            if force:
+                args = ["install", "--upgrade", "--force-reinstall", "--no-deps",
+                        os.path.join(dir, package)]
+                args[0:0] = pip
+                args = add_test_params(args)
+                exit_code = call(args, stdout=sys.stdout, stderr=sys.stderr)
+                if exit_code > 0:
+                    raise Exception("execute %s error! exit code: %d" % (" ".join(args),
+                                                                         exit_code))
+                args = ["install", os.path.join(dir, package)]
+                args[0:0] = pip
+                args = add_test_params(args)
+                exit_code = call(args, stdout=sys.stdout, stderr=sys.stderr)
+                if exit_code > 0:
+                    raise Exception("execute %s error! exit code: %d" % (" ".join(args),
+                                                                         exit_code))
+            else:
+                args = ["install", os.path.join(dir, package)]
+                args[0:0] = pip
+                args = add_test_params(args)
+                exit_code = call(args, stdout=sys.stdout, stderr=sys.stderr)
+                if exit_code > 0:
+                    raise Exception("execute %s error! exit code: %d" % (" ".join(args),
+                                                                         exit_code))
+            return exit_code
+    if optional:
+        return 0
+    raise Exception("package '" + name + "' not found")
+
+
+# try to install user-provided apache beam sdk
+sdk_whl_file = find_beam_sdk_whl()
+if sdk_whl_file != "":
+    try:
+        exit_code = pip_install_package(files, staged_dir, sdk_whl_file, False, False, ["gcp"])
+    except Exception as e:
+        print(e)
+
+# try to install the user-provided dataflow python sdk
+sdk_src_file = "dataflow_python_sdk.tar"
```

Comment:
dataflow package is not needed and should not be installed

Reply:
Removed.